examples/todo/src/datasources/db.datasource.ts

```diff
@@ -3,22 +3,11 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import * as path from 'path';
-// The juggler reference must exist for consuming code to correctly infer
-// type info used in the "db" export (contained in juggler.DataSource).
-// tslint:disable-next-line:no-unused-variable
-import {juggler} from '@loopback/repository';
+import {inject} from '@loopback/core';
+import {juggler, DataSource} from '@loopback/repository';
 
-const dsConfigPath = path.resolve(
-  __dirname,
-  '../../../config/datasources.json',
-);
-const config = require(dsConfigPath);
-
-// TODO(bajtos) Ideally, datasources should be created by @loopback/boot
-// and registered with the app for dependency injection.
-// However, we need to investigate how to access these datasources from
-// integration tests where we don't have access to the full app object.
-// For example, @loopback/boot can provide a helper function for
-// performing a partial boot that creates datasources only.
```

Comment:
Could you please clarify how integration tests can access datasource configuration now? I guess the answer could be found somewhere in the code, but it will be easier for reviewers if you could summarize it in a comment.

Reply:
One can access the config / DataSource by getting it from the context. I'm not sure of the question here ...

Reply:
In integration tests, there is no Application/Context object. We want to test the integration between a repository and the backing database, not the integration with the app/context.

Also test-data-builders need to access a repository instance directly, see e.g. http://loopback.io/doc/en/lb4/Implementing-features.html#update-test-helpers-and-the-controller-use-real-model-and-repository

```ts
async function givenEmptyDatabase() {
  await new ProductRepository().deleteAll();
}

async function givenProduct(data: Partial<Product>) {
  return await new ProductRepository().create(
    Object.assign(
      {
        name: 'a-product-name',
        slug: 'a-product-slug',
        price: 1,
        description: 'a-product-description',
        available: true,
      },
      data,
    ),
  );
}
```

Reply:
For the scope of this pull request,  it's enough to allow tests to instantiate `DbDataSource` without the need to create a custom datasource config:

```ts
const db = new DbDataSource();
```

This can be achieved e.g. by adding a default value for `dsConfig` argument, as discussed and shown in https://github.com/strongloop/loopback-next/pull/1385#discussion_r193226113.

I created a follow-up issue to improve the overall design to allow tests to modify the default datasource configuration, see https://github.com/strongloop/loopback-next/issues/1396